pkg/operator/bootstrap.go

```diff
@@ -24,44 +24,57 @@ const randomSecretSize = 64
 func (h *Handler) Bootstrap() (*regopapi.ImageRegistry, error) {
 	// TODO(legion): Add real bootstrap based on global ConfigMap or something.
 
-	cr := &regopapi.ImageRegistry{
+	crList := &regopapi.ImageRegistryList{
 		TypeMeta: metav1.TypeMeta{
 			APIVersion: regopapi.SchemeGroupVersion.String(),
 			Kind:       "ImageRegistry",
 		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "image-registry",
-			Namespace: h.params.Deployment.Namespace,
-		},
 	}
 
-	err := sdk.Get(cr)
+	err := sdk.List(h.params.Deployment.Namespace, crList)
 	if err != nil {
 		if !errors.IsNotFound(err) {
-			return nil, fmt.Errorf("failed to get image-registry custom resource: %s", err)
+			return nil, fmt.Errorf("failed to list registry custom resources: %s", err)
 		}
-	} else {
-		return cr, nil
+	}
+
+	switch len(crList.Items) {
+	case 0:
+		// let's create it.
+	case 1:
+		return &crList.Items[0], nil
+	default:
+		return nil, fmt.Errorf("only one registry custom resource expected in %s namespace, got %d", h.params.Deployment.Namespace, len(crList.Items))
 	}
 
 	logrus.Infof("generating registry custom resource")
 
-	cr.Spec = regopapi.ImageRegistrySpec{
-		OperatorSpec: operatorapi.OperatorSpec{
-			ManagementState: operatorapi.Managed,
-			Version:         "none",
-			ImagePullSpec:   "docker.io/openshift/origin-docker-registry",
+	cr := &regopapi.ImageRegistry{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: regopapi.SchemeGroupVersion.String(),
+			Kind:       "ImageRegistry",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "image-registry",
+			Namespace: h.params.Deployment.Namespace,
 		},
-		Storage: regopapi.ImageRegistryConfigStorage{
-			Filesystem: &regopapi.ImageRegistryConfigStorageFilesystem{
-				VolumeSource: corev1.VolumeSource{
-					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-						ClaimName: "image-registry-pvc",
+		Spec: regopapi.ImageRegistrySpec{
+			OperatorSpec: operatorapi.OperatorSpec{
+				ManagementState: operatorapi.Managed,
+				Version:         "none",
+				ImagePullSpec:   "docker.io/openshift/origin-docker-registry",
+			},
+			Storage: regopapi.ImageRegistryConfigStorage{
```

Comment:
so as discussed this morning, please change this to an empty configstorage and make sure the operator reports a reasonable error when it sees an empty configstorage.

And when we actually have enough information to make a good decision about what the initial storage configuration should be, we'll do that.

Reply:
(you can do that in a followup, I don't want to confuse this PR)